File: angular-app/src/app/ddd-core/application/decorator/helpers/strategies/abstracts/base-helper-strategy.ts
import { UserAction } from '../../../../enums/user-action';
import { DomainConstructor } from '../../../../strategies/abstracts/base-application-strategy';

export abstract class BaseHelperStrategy {
  protected priority = 0;

  abstract canHandle(normalized: string, tokens: string[]): boolean;

  abstract handle(normalized: string, tokens: string[]): [UserAction, string | DomainConstructor | null];
}

File: python-app/application/decorator/strategies/abstracts/base_helper_strategy.py
from abc import ABC, abstractmethod
from collections.abc import Sequence
from typing import Any
from application.enums.user_action import UserAction


class BaseHelperStrategy(ABC):
    _priority: int = 0

    @abstractmethod
    def can_handle(self, normalized: str, tokens: Sequence[str]) -> bool:
        pass

    @abstractmethod
    def handle(self, normalized: str, tokens: Sequence[str]) -> tuple[UserAction, Any | None]:
        pass
